libs/npm/core/src/utils/elastic/generateQuery.ts

```diff
@@ -1,68 +1,30 @@
+import { MsearchMultisearchBody } from '@elastic/elasticsearch/lib/api/types';
+
 import { IFilterSettingsModel } from '../../hooks';
-import { generatePublishedOnQuery } from './generatePublishedOnQuery';
-import { generateQueryForActions } from './generateQueryForActions';
-import { generateQueryForExistCheck } from './generateQueryForExistCheck';
-import { generateRangeForArrayField } from './generateRangeForArrayField';
-import { generateTerm } from './generateTerm';
-import { generateTerms } from './generateTerms';
-import { generateTermsForArrayField } from './generateTermsForArrayField';
-import { generateTextQuery } from './generateTextQuery';
+import { generateFilterQuery } from './generateFilterQuery';
+import { generateMustNotQuery } from './generateMustNotQuery';
+import { generateMustQuery } from './generateMustQuery';
 
 /**
  * Generates an Elasticsearch query based on specified 'query'.
  * @param settings Form values that will be used to configure the elasticsearch query.
  * @param query Original query object.
  * @returns Elasticsearch query JSON.
  */
-export const generateQuery = (settings: IFilterSettingsModel, query: any = {}) => {
-  var elastic = { ...query };
+export const generateQuery = (
+  settings: IFilterSettingsModel,
+  query: any = {},
+  condition: 'must' | 'must_not' | 'filter' = 'must',
+): MsearchMultisearchBody => {
+  var elastic = query;
+  if (condition === 'must') elastic = generateMustQuery(settings, query);
```

Comment:
Moved most of the logic out of this into separate functions to remove duplication.